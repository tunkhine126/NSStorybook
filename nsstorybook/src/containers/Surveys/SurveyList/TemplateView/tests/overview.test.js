/**
 *
 * Tests for Overview
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import TestingComponent from 'components/shared/Testing';

import Overview from '../overview';

describe('<Overview />', () => {
  const props = {
    data: [],
    submissionData: [],
    userContext: 'us_en',
    match: {
      isExact: true,
      params: { id: '9f11cc4a-0baa-11ea-b9a3-22000acba051' },
      path: '/surveys/survey-templates/:id',
      url: '/surveys/survey-templates/9f11cc4a-0baa-11ea-b9a3-22000acba051',
    },
  };

  it('Expect no errors rending Overview', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <Overview {...props} />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <Overview {...props} />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
