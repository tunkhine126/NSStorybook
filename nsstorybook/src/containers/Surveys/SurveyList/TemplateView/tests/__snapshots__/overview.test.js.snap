// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-11 MuiGrid-grid-lg-8"
  >
    <div
      class="MuiPaper-root makeStyles-table-204 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="survey overview table"
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Questions
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Languages
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Submissions
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Last Submission
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              0
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  0
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root makeStyles-root-363 MuiButton-text makeStyles-btn-359 makeStyles-primary-360 makeStyles-viewBtn-206"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label makeStyles-label-364"
                    >
                      View
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  N/A
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
